File: Practice/11/Python/exponentiation.py
#!/usr/bin/python3

# Задание 11 - Возведение в степень

print("11. Возведение в степень")


def piwpow(base, power):
    """Caoution! This is fucking recursion!"""
    if power == 0:
        return 1
    else:
        return base if power == 1 else base * piwpow(base, power - 1)


print(piwpow(float(input('Основание: ')),
             int(input('Степень: '))))

File: Practice/11/C++/exponentiation.cpp
#include <iostream>

/* Задание 11 - Возведение в степень */

using namespace std;

float piwpow(float base, int power)
/* Caoution! This is fucking recursion! */
{
    return power == 1 ? base : base * piwpow(base, power - 1);
}

int main()
{
    cout << "11. Возведение в степень" << endl;
    float base;
    int power;
    cout << "Основание: ";
    cin >> base;
    cout << "Степень: ";
    cin >> power;

    cout << piwpow(base, power) << endl;
    return 0;
}
